<a href="https://colab.research.google.com/github/nourswaid/ML-A_Z-/blob/main/Classification/full_project/BreastCancerDetection.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv('/content/Data (1).csv')
x=df.iloc[:,:-1].values
y=df.iloc[:,-1].values


In [3]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,random_state=0,test_size=0.2)

In [4]:
x_train

array([[ 752904,      10,       1, ...,       5,       4,       1],
       [1218860,       1,       1, ...,       3,       1,       1],
       [ 411453,       5,       1, ...,       3,       1,       1],
       ...,
       [1214092,       1,       1, ...,       1,       1,       1],
       [1303489,       3,       1, ...,       2,       1,       1],
       [ 378275,      10,       9, ...,       7,       7,       1]])

In [5]:
from sklearn.preprocessing import StandardScaler
sc=StandardScaler()
x_train=sc.fit_transform(x_train)
x_test=sc.transform(x_test)

In [10]:
#logistic

from sklearn.linear_model import LogisticRegression
le=LogisticRegression(random_state=0)
le.fit(x_train,y_train)

prediction=le.predict(x_test)

In [11]:
from sklearn.metrics import confusion_matrix, accuracy_score

cm=confusion_matrix(y_test,prediction)
print(accuracy_score(y_test,prediction))
print(cm)


0.9562043795620438
[[84  3]
 [ 3 47]]


In [14]:
#knn

from sklearn.neighbors import KNeighborsClassifier
knn=KNeighborsClassifier(n_neighbors=5)
knn.fit(x_train,y_train)

prediction=knn.predict(x_test)

In [15]:
from sklearn.metrics import confusion_matrix, accuracy_score

cm=confusion_matrix(y_test,prediction)
print(accuracy_score(y_test,prediction))
print(cm)


0.9562043795620438
[[83  4]
 [ 2 48]]


In [16]:
from sklearn.svm import SVC
sv=SVC(kernel='rbf',random_state=0)
sv.fit(x_train,y_train)

prediction=sv.predict(x_test)

In [17]:
from sklearn.metrics import confusion_matrix, accuracy_score

cm=confusion_matrix(y_test,prediction)
print(accuracy_score(y_test,prediction))
print(cm)

0.9562043795620438
[[82  5]
 [ 1 49]]


In [18]:
#linear svm
from sklearn.svm import SVC
sv=SVC(kernel='linear',random_state=0)
sv.fit(x_train,y_train)

prediction=sv.predict(x_test)

In [19]:
cm=confusion_matrix(y_test,prediction)
print(accuracy_score(y_test,prediction))
print(cm)

0.9562043795620438
[[83  4]
 [ 2 48]]


In [20]:
# naive

from sklearn.naive_bayes import GaussianNB
nb=GaussianNB()
nb.fit(x_train,y_train)

prediction=nb.predict(x_test)

In [21]:
cm=confusion_matrix(y_test,prediction)
print(accuracy_score(y_test,prediction))
print(cm)

0.948905109489051
[[80  7]
 [ 0 50]]


In [23]:
# decesion tree

from sklearn.tree import DecisionTreeClassifier
ds=DecisionTreeClassifier(criterion='entropy',random_state=0)
ds.fit(x_train,y_train)

prediction=ds.predict(x_test)

In [24]:
cm=confusion_matrix(y_test,prediction)
print(accuracy_score(y_test,prediction))
print(cm)

0.9708029197080292
[[85  2]
 [ 2 48]]


In [25]:
from sklearn.ensemble import RandomForestClassifier
cl=RandomForestClassifier(n_estimators=10,criterion='entropy',random_state=0)
cl.fit(x_train,y_train)

prediction=cl.predict(x_test)

In [26]:
cm=confusion_matrix(y_test,prediction)
print(accuracy_score(y_test,prediction))
print(cm)

0.9562043795620438
[[84  3]
 [ 3 47]]


## Conclusion
## Decesion tree classifier model got the best accracy of 0.97080



